rfc/composite_table_availability.pseudocode

```diff
@@ -0,0 +1,116 @@
+
+PhysicalTableSchema:
+	getFieldName(Column c)
+
+DataSource:
```

Comment:
I think DataSourceMetadata has all the state we need for this interface, so we should repurpose it for implementation, yes.